### PR TITLE
Added in missing comments for authentication

### PR DESCRIPTION
**Pull request by jadahiya-MSFT**

### CODE REVIEW

File: packages/teams-js/src/public/authentication.ts [hunk 1]
```diff
@@ -482,6 +484,9 @@ export namespace authentication {
     failureCallback?: (reason: string) => void;
   }
 
+  /**
+   * An interface that describes the Authentication pop up's parameters
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:488]: How about: 
Describes Authentication pop-up parameters

File: packages/teams-js/src/public/authentication.ts [hunk 2]
```diff
@@ -507,6 +512,9 @@ export namespace authentication {
    */
   export type AuthenticateParameters = AuthenticatePopUpParameters & LegacyCallBacks;
 
+  /**
+   * An interface that describes the Authentication Token's Request Parameters
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:516]: Describes authentication token request parameters

**Review by erikadoyle — COMMENTED** [review]
Looks good, just some minor things

**Review by erinha — APPROVED** [review]

File: packages/teams-js/src/public/authentication.ts [hunk 3]
```diff
@@ -466,6 +466,8 @@ export namespace authentication {
   /**
    * @deprecated
    * As of 2.0.0-beta.1.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:468]: Sorry! Just realized this--the @deprecated tag generates a full sentence at publish time: [This API is now deprecated](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/authentication.legacycallbacks?view=msteams-client-js-beta)
. . . so it kind of strands "As of 2.0.0.-beta.1." on its own line without much context (see link above for example). Could we rework each instance of "As of 2.0.0.-beta.1." in this file? Also @erinha correct me if I'm wrong, but I think we landed on dropping the "-beta.1" from the deprecated version note. 

How about something like: **As of 2.0.0, this [function|interface] has been deprecated in favor of a Promise-based model.**

Comment (jadahiya-MSFT) [reply packages/teams-js/src/public/authentication.ts:468]: Got it, I'll make that change

Comment (erikadoyle) [reply packages/teams-js/src/public/authentication.ts:468]: Awesome, thanks! Everything else looks good.

**Review by TrevorJoelHarris — APPROVED** [review]

File: packages/teams-js/src/public/authentication.ts (hunk 3 above)
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/authentication.ts]: @erikadoyle if you are able to approve this PR, please feel free to do that when you're happy with the updated comments!

**Review by shrshindeMSFT — APPROVED** [review]

**Pull request merged**